File: arduinobot_ws/src/arduinobot_test/scripts/simple_publisher.py
#!/usr/bin/env python3
import rospy
from std_msgs.msg import String

if __name__ == '__main__':
    # Inizialize a ROS node called talker
    rospy.init_node('simple_publisher_py', anonymous=True)

    # register a publisher on the topic /chatter that will publish String messages
    pub = rospy.Publisher('chatter', String, queue_size=10)
        
    # Define the frequency for publishing the messages
    # The rate is expressed in Hertz
    rate = rospy.Rate(10) # 10hz
    counter = 0

    # Keep going publishing messages until the ROS communication is alive
    while not rospy.is_shutdown():
        hello_msg = "hello world %d" % counter
        pub.publish(hello_msg)
        # wait the desired rate before publishing the next message
        rate.sleep()
        counter += 1

File: arduinobot_ws/src/arduinobot_test/src/simple_publisher.cpp
#include "ros/ros.h"
#include "std_msgs/String.h"
#include <sstream>


int main(int argc, char **argv)
{
    // Inizialize a ROS node called talker
    ros::init(argc, argv, "simple_publisher_cpp");
    ros::NodeHandle n;

    // register a publisher on the topic /chatter that will publish String messages
    ros::Publisher pub = n.advertise<std_msgs::String>("chatter", 10);

    // Define the frequency for publishing the messages
    // The rate is expressed in Hertz
    ros::Rate rate(10); // Hz


    int counter = 0;
    // Keep going publishing messages until the ROS communication is alive
    while (ros::ok())
    {
        std_msgs::String msg;

        std::stringstream ss;
        ss << "Hello world " << counter;
        msg.data = ss.str();
        pub.publish(msg);

        ros::spinOnce();

        // wait the desired rate before publishing the next message
        rate.sleep();
        ++counter;
    }


  return 0;
}